File: Unity/TSProj/src/mods/ModTheSpire/Scripts/Power/MinionPower.ts
import FileHelper from "../FileHelper";
import { LocalizationProperty } from "../Gen/DB/Localization";
import { AbstractPower, PowerType } from "./AbstractPower";

export class MinionPower extends AbstractPower{
    Icon: string = FileHelper.FormatPath("Powers/minion.png");
    Id: string = "Minion";
    Name: string = LocalizationProperty.Read("能力-爪牙");
    get Type(): PowerType {
        return PowerType.Buff;
    }

    GetDescription(...args: any[]): string {
        return LocalizationProperty.Read("能力-爪牙描述");
    }
}

File: Unity/Assets/Scripts/ModTheSpire/Power/Special/MinionPower.cs
using TCG.Config;

namespace ModTheSpire.Power.Special
{
    public class MinionPower : AbstractPower
    {
        public override string Id { get; } = nameof(MinionPower);
        public override string Name { get; } = LocalizationProperty.Read("能力-爪牙");
        public override string Icon { get; } = "Powers/minion.png";
        public override PowerType Type { get; } = PowerType.Special;
        public override string GetDescription(object[] args)
        {
            return LocalizationProperty.Read("能力-爪牙描述");
        }
    }
}